src/api/app/controllers/webui/request_controller.rb

```diff
@@ -212,6 +212,30 @@ def request_action
     end
   end
 
+  def request_action_changes
+    # TODO: Change @diff_limit to a local variable
+    @diff_limit = params[:full_diff] ? 0 : nil
+    # TODO: Change @actions to a local variable
+    @actions = @bs_request.webui_actions(filelimit: @diff_limit, tarlimit: @diff_limit, diff_to_superseded: @diff_to_superseded, diffs: true,
```

Comment:
you already specify the `action_id` here, there should be no need to do `@actions.find { |action| action[:id] == params['id'].to_i }` again in the next line, you can simply call `.first` on it

Reply:
Yes. The code can be refactored and adapted to what we already have in the show action of the request controller, regarding retrieving actions and a specific action.

But we opted for fixing the code, reverting it to what was already working. and modifying it as less as possible. We wrote down all those TODO's not to forget where we could improve the code. Basically, we first make it work, and then we refactor. It could be confusing to mix fixing and refactoring in the same PR.